app/tasks/maintenance/t20241127_migrate_champ_expression_reguliere_to_formatted_task.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Maintenance
+  class T20241127MigrateChampExpressionReguliereToFormattedTask < MaintenanceTasks::Task
+    # Documentation: les champs ExpressionReguliere sont migrés au nouveau champ Formatted
+    # en mode avancé.
+    # Voir aussi la maintenance task de transformation des types de champs.
+
+    include RunnableOnDeployConcern
+    include StatementsHelpersConcern
+
+    # Uncomment only if this task MUST run imperatively on its first deployment.
+    # If possible, leave commented for manual execution later.
+    run_on_first_deploy
+
+    def collection
+      with_statement_timeout("5min") do
+        Champ.where(type: ["Champs::ExpressionReguliereChamp", "Champs::FormattedChamp"]).in_batches
+      end
+    end
+
+    def process(batch)
+      batch.update_all(type: "Champs::FormattedChamp")
+    end
```

Comment:
a priori ça marchera pas comme ça et ça fera un timeout.
Ce qu'on fait ailleurs   donnerait : 

```suggestion
    def collection
      with_statement_timeout("15min") do
        Champ.where(type: ["Champs::ExpressionReguliereChamp", "Champs::FormattedChamp"])
      end
    end

    def process(champ)
      champ(type: "Champs::FormattedChamp")
    end
    
    def count = nil # sinon ça timeout
```

pour rappel le fait d'intégrer `"Champs::FormattedChamp" `à la collection permet de ne pas changer le scope et donc de ne pas "rater" des champs à cause de la pagination.